### PR TITLE
Add content manager API contracts

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Add API contracts to the Content Manager package

### CODE REVIEW

**Comment by jhoward1994** [comment]
@Marc-Roig the contracts still need to be written but I just wanted to check this file structure with you. As there is no admin I thought it didnt make sense to make a `shared` folder. wdyt?

**Comment by joshuaellis** [comment]
I agree @Marc-Roig try keep it consistent with other package structures. You'll also probably want to `pack-up` the shared stuff anyway cause you've written `.ts` instead of `.d.ts` 🤔

File: packages/core/content-manager/shared/contracts/collection-types.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+  }
+
+  export interface Response {
+    data: Entity;
+    error?: errors.ApplicationError;
+  }
+}
+
+/**
+ * POST /collection-types/:model/:id/actions/publish
```
Comment (joshuaellis) [inline packages/core/content-manager/shared/contracts/collection-types.ts:162]: This format is very nice

**Review by Marc-Roig — APPROVED** [review]
🔥

**Pull request merged**